src/main/java/com/google/javascript/gents/TypeScriptGenerator.java

```diff
@@ -176,26 +183,42 @@ private void generateTypeScript() {
 
     new StyleFixPass(compiler, comments).process(externRoot, srcRoot);
 
+    new AstValidator(
+            compiler,
+            new AstValidator.ViolationHandler() {
+              @Override
+              public void handleViolation(String message, Node n) {
+                compiler.report(JSError.make(n, CLUTZ_MALFORMED_AST, message));
```

Comment:
Thanks for trying this! I've tried to add source location to places where we create new nodes, but I'm sure there are lots of places missing – the API doesn't make this easy retroactively.